pkg/askgit/askgit.go

```diff
@@ -140,45 +128,21 @@ func (a *AskGit) RepoPath() string {
 	return a.options.RepoPath
 }
 
-func (a *AskGit) loadGitQLiteModules(conn *sqlite3.SQLiteConn) error {
-	_, err := exec.LookPath("git")
-	localGitExists := err == nil
-
-	if !a.options.UseGitCLI || !localGitExists {
-		err = conn.CreateModule("commits", gitqlite.NewGitLogModule(&gitqlite.GitLogModuleOptions{RepoPath: a.RepoPath()}))
-		if err != nil {
-			return err
-		}
-	} else {
-		err = conn.CreateModule("commits", gitqlite.NewGitLogCLIModule(&gitqlite.GitLogCLIModuleOptions{RepoPath: a.RepoPath()}))
-		if err != nil {
-			return err
-		}
-	}
-
-	err = conn.CreateModule("stats", gitqlite.NewGitStatsModule(&gitqlite.GitStatsModuleOptions{RepoPath: a.RepoPath()}))
-	if err != nil {
-		return err
-	}
-
-	err = conn.CreateModule("files", gitqlite.NewGitFilesModule(&gitqlite.GitFilesModuleOptions{RepoPath: a.RepoPath()}))
-	if err != nil {
-		return err
-	}
-
-	err = conn.CreateModule("tags", gitqlite.NewGitTagsModule(&gitqlite.GitTagsModuleOptions{RepoPath: a.RepoPath()}))
-	if err != nil {
-		return err
-	}
-
-	err = conn.CreateModule("branches", gitqlite.NewGitBranchesModule(&gitqlite.GitBranchesModuleOptions{RepoPath: a.RepoPath()}))
-	if err != nil {
-		return err
+func (a *AskGit) loadGitModules(conn *sqlite3.SQLiteConn) error {
+	var sp = fmt.Sprintf
+	var stmts = []string{
+		sp("CREATE VIRTUAL TABLE commits  USING git_log(path=%q)", a.RepoPath()),
+		sp("CREATE VIRTUAL TABLE branches USING git_branches(path=%q)", a.RepoPath()),
+		sp("CREATE VIRTUAL TABLE blame    USING git_blame(path=%q)", a.RepoPath()),
+		sp("CREATE VIRTUAL TABLE files    USING git_files(path=%q)", a.RepoPath()),
+		sp("CREATE VIRTUAL TABLE stats    USING git_stats(path=%q)", a.RepoPath()),
+		sp("CREATE VIRTUAL TABLE tags     USING git_tags(path=%q)", a.RepoPath()),
```

Comment:
one thing to note, functionally it makes no difference I believe (in this case), but it looks like these new tables are loaded in as "regular" virtual tables (where `CREATE VIRTUAL TABLE...` is necessary), previously they were ["eponymous-only" virtual tables](https://sqlite.org/vtab.html#eponymous_only_virtual_tables), but did not take any parameters. In future work, when these git tables use the `vtab` library, I think they will likely return to being "eponymous-only" as this would allow us to parameterize the path to the git repo `SELECT * FROM git_log('/path/to/some/repo')` as well as other options/filters (for instance, like a `commit_id` param to the `files` table).

Nothing to change here, just sharing thoughts

Reply:
yep right! we can revert back to eponymous later when we have things like multi-repo support etc.